SplitFiles/customer.c

```diff
@@ -0,0 +1,168 @@
+//////////////////////////////////////////////////////////
+//
+// Customers
+//
+//////////////////////////////////////////////////////////
+
+// Can be compiled with:
+// zcc +zx -vn -startup=1 -clib=sdcc_iy -D_TEST_CUSTOMER bread.c util.c customer.c -o customer -create-app
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "customer.h"
+#include "bread.h"
+#include "util.h"
+
+void redraw_customers(CustomerBase *base) {
+    unsigned char ypos = 4;
+    Customer* lastCustomer = base->rootCustomer;
+    while (lastCustomer->nextCustomer != NULL) {
+        lastCustomer = lastCustomer->nextCustomer;
+        printf(PRINTAT"\x12%c""%-12s", ypos, lastCustomer->breadOrder->desc);
+        ++ypos;
+    }
+    for (; ypos < 4 + MAX_CUSTOMER_COUNT; ++ypos) {
+        printf(PRINTAT"\x12%c""%-12s", ypos, "");
+    }
+}
+
+void customer_func(GameComponent* customers, GameParameters* params) {
+    CustomerBase* base = (CustomerBase*)(customers->ptr);
+    Customer* newCustomer = (Customer*)NULL;
+    int redraw = 0;
+
+    Customer* thisCustomer;
+    Customer* lastCustomer;
+
+    if (params->messageAddress == 20) {
+        //Someone sent toast!
+        BreadState* bread = (BreadState*)params->message;
+        thisCustomer = base->rootCustomer->nextCustomer;
+        lastCustomer = base->rootCustomer;
+        while (thisCustomer != NULL) {
+            if (bread->type->letter == thisCustomer->breadOrder->letter) {
```

Comment:
Not an issue about multiple customers ordering the same? Or are they always in customer order?

Reply:
Earliest customer who asked for that type of toast gets it - assuming there is someone